library/psa_crypto_slot_management.c

```diff
@@ -145,10 +145,13 @@ psa_status_t psa_initialize_key_slots(void)
      * means that all the key slots are in a valid, empty state.
      * If multi-threading is enabled, then initialize the
      * global key slot mutex. */
-    global_data.key_slots_initialized = 1;
 #if defined(MBEDTLS_THREADING_C)
-    mbedtls_mutex_init(&global_data.key_slot_mutex);
+    if(!global_data.key_slots_initialized) {
```

Comment:
I think we need a comment here

```
/* XXX not thread safe */
```

I have a "once" implementation (with versions for pthreads, Apple's GCD, and Windows, as well as an "unthreaded" version) that we can use to fix this, but we should note the issue